src/gui/editorwidgets/qgsexternalresourcewidgetwrapper.cpp

```diff
@@ -79,15 +79,29 @@ bool QgsExternalResourceWidgetWrapper::valid() const
 
 void QgsExternalResourceWidgetWrapper::setFeature( const QgsFeature &feature )
 {
-  if ( mQgsWidget && mPropertyCollection.isActive( QgsEditorWidgetWrapper::RootPath ) )
+  if ( mQgsWidget )
   {
-    QgsExpressionContext expressionContext( QgsExpressionContextUtils::globalProjectLayerScopes( layer() ) );
-    expressionContext.setFeature( feature );
-    bool ok = false;
-    QString path = mPropertyCollection.valueAsString( QgsEditorWidgetWrapper::RootPath, expressionContext, QString(), &ok );
-    if ( ok )
-    {
-      mQgsWidget->setDefaultRoot( path );
+    if ( mPropertyCollection.isActive( QgsEditorWidgetWrapper::RootPath ) )
+    {
+      QgsExpressionContext expressionContext( QgsExpressionContextUtils::globalProjectLayerScopes( layer() ) );
```

Comment:
I'd suggest moving the construction of the expression context up from this scope so that the same context can be reused by both properties